### ISSUE TITLE
`paren_brace_linter` false positive within regex strings: "(match_group){quantifier}"

### ISSUE THREAD

**Issue by russHyde** [jimhester/lintr#441]
`paren_brace_linter` flags the "){" within regex definition below:

```
#' Checks if a file has a .tar or .tar.bz2 extension
#'
#' @noRd
.is_tar_file <- function(x) {
  grepl(pattern = ".*\\.tar(\\.bz2){0,1}", x)
}
```

Should `paren_brace_linter` be made more strict - so that it doesn't consider strings?

**Comment by Fernal73** [comment]
```R
grepl("(iss){2}", "Mississippi")
```

**Comment by russHyde** [comment]
This seems to be a more general issue with using regex-based linters.

For example, the following throws with `equals_na_linter`, despite not being part of the active code
```
dont_lint_me <- "xyz == NA shouldn't occur in an R script"

x <- NA
dont_lint_me_either <- is.na(x) # use is.na instead of x == NA
```

**Comment by Fernal73** [comment]
This can be mitigated by allowing the # nolint comment to specify the linter to be turned off.
eg,
#nolint paren_brace_linter

**Comment by bersbersbers** [comment]
It seems this did not make it into a release: it was fixed on Dec 4, 2020, and there was a 2.0.1 release on Dev 5, 2020: https://github.com/jimhester/lintr/releases/tag/v2.0.1

However, 2.0.1 on CRAN is from Feb 19, *2020*: https://cran.r-project.org/web/packages/lintr/index.html

**Comment by russHyde** [comment]
I believe  the commit that was tagged as v2.0.1 came from Feb 2020 (https://github.com/jimhester/lintr/commit/01e57c2124891b965c02626b27559c95f3218679). It's simply that the released 2.0.1 version didn't get tagged on github at the time that it was released.

**Comment by bersbersbers** [comment]
Sounds convincing, but that means there hasn't been a release in over a year. Would you recommend installing from GitHub master rather than from CRAN?

**Comment by russHyde** [comment]
If having this paren-brace issue solved is necessary, then use the github master. I always use the github master in continuous integration. Locally, I use multiple R environments: using the CRAN-lintr when working on stuff other than lintr.
